blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java

```diff
@@ -234,7 +232,22 @@ public HttpResponse submitInputStep(StaplerRequest request) {
             }
 
             //XXX: execution.doProceed(request) expects submitted form, otherwise we could have simply used it
-            execution.preSubmissionCheck();
+
+            try {
+                execution.preSubmissionCheck();
+            } catch (Failure f) {
+                JSONObject obj = new JSONObject();
```

Comment:
Backend error messages are in specific format. Lets keep it consistent. Following code should do it:

```java
...
rsp.getWriter().print(JsonConverter.toJson(new ErrorMessage(400, f.getMessage())));
....
```